WordPress/openverse-catalog
# Generate TSV filenames in separate step

## Fixes


Fixes #619 by @AetherUnbound

## Description


This PR adds a separate task at the very beginning of the provider DAG factory which generates the TSV filenames up front. The pull data task now pulls the generated filenames and sets the appropriate store output path(s) prior to execution.

This should help us with #549 and #618 I believe! Ultimately I believe we should be able to completely alter our approach to TSV filenames down the road once we have all the provider scripts unified.

In doing this refactor, I split some functions out into a `factory_utils.py` module. I'm not sold on the name, so let me know if you think of any other names :slightly_smiling_face: I also moved some test files around to reflect their current position in the file tree.

### Screenshots
**Image DAG**
![Screenshot_2022-07-21_14-32-52](https://user-images.githubusercontent.com/10214785/180319903-80ca79b5-5993-44a4-94a2-d3d60e5ded52.png)

**Audio DAG**
![Screenshot_2022-07-21_14-32-30](https://user-images.githubusercontent.com/10214785/180319905-b0f9d40f-07ae-41d4-b757-45871aa5156f.png)

**Mixed DAG**
![Screenshot_2022-07-21_14-32-40](https://user-images.githubusercontent.com/10214785/180319904-8b655b86-120b-49d4-b27f-2b755cf51430.png)

## Testing Instructions

Run an image DAG, and audio DAG, and the wikimedia DAG and ensure that each of the `copy_to_s3` and `pull_media` steps are using the filenames generated in generate TSV filenames step. The "Rendered" tab for a certain task in the UI can help quickly determine what values the downstream tasks received for the filenames.
![Screenshot_2022-07-21_14-38-36](https://user-images.githubusercontent.com/10214785/180319998-b5be9bbe-4b01-4521-9c7c-a6c2517ba983.png)




## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
